symengine/series_generic.cpp

```
@@ -2,155 +2,242 @@
+    if (s.get_univariate_poly()->get_var() != symbol("x"))
+        throw std::runtime_error("Polynomial is not x");
+    if (n < 0)
+        throw std::runtime_error("Not Implemented");
+    
+    if (n == 0) {
+        return UnivariateExprPolynomial(1);
+    } else if (n == 1) {
+        return UnivariateSeries::trunc_poly(s.get_univariate_poly()->get_var(), s.get_univariate_poly()->get_dict(), prec);
+    } else {
+        UnivariateExprPolynomial p = s;
+        for(int i = 0; i < n; i++) {
+            p = UnivariateSeries::mul(s, p, prec);
+        }
+        return p;
     }
-    return make_rcp<const UnivariateSeries>(a.var_, minprec, max, std::move(dict));
 }
 
-RCP<const UnivariateSeries> neg_uni_series (const UnivariateSeries& a)
-{
-    return make_rcp<const UnivariateSeries>(a.var_, a.prec_, std::move(neg_poly(*a.poly_)));
+Expression UnivariateSeries::find_cf(const UnivariateExprPolynomial &s, const UnivariateExprPolynomial &var, unsigned deg) {
+    return (s.get_univariate_poly()->get_dict()).at(deg);
+}
+
+Expression UnivariateSeries::root(Expression &c, unsigned n) {
+    return pow_ex(c, 1/Expression(n));
+}
+
+UnivariateExprPolynomial UnivariateSeries::diff(const UnivariateExprPolynomial &s, const UnivariateExprPolynomial &var) {
+    RCP<const Basic> p = s.get_univariate_poly()->diff(var.get_univariate_poly()->get_var());
+    if (is_a<const UnivariatePolynomial>(*p))
+        return UnivariateExprPolynomial(rcp_static_cast<const UnivariatePolynomial>(p));
+    else
+        throw std::runtime_error("Not a UnivariatePolynomial");
 }
 
-RCP<const UnivariateSeries> sub_uni_series (const UnivariateSeries& a, const UnivariateSeries& b)
-{
+UnivariateExprPolynomial UnivariateSeries::integrate(const UnivariateExprPolynomial &s, const UnivariateExprPolynomial &var) {
+    map_int_Expr dict;
+    for (auto &it : s.get_univariate_poly()->get_dict())
+        dict.insert(std::pair<int, Expression>(it.first + 1, it.second / (it.first + 1)));
```

You should throw an error if `it.first` == -1